yarn-project/circuit-types/src/tx/tx.ts

```diff
@@ -165,6 +165,18 @@ export class Tx {
 
       proofSize: this.proof.buffer.length,
       size: this.toBuffer().length,
+
+      feePaymentMethod:
+        // needsTeardown? then we pay a fee
+        this.data.needsTeardown
+          ? // needsSetup? then we pay through a fee payment contract
+            this.data.needsSetup
+            ? // if the first call is to `approve_public_authwit`, then it's a public payment
```

Comment:
Will we make this assumption on mainnet? Just wondering.

Reply:
It's a weak assumption. I think it's ok for benchmarking right now, but will have to be changed later. WDYT?

Reply:
I agree. Fine for now, but we probably want some clarity, first for ourselves, then for other sequencers on how to interpret what is going on in setup and teardown.